core/src/impl/Kokkos_HostSpace_deepcopy.hpp

```diff
@@ -23,7 +23,7 @@ namespace Kokkos {
 
 namespace Impl {
 
-void hostspace_fence(const DefaultHostExecutionSpace& exec);
+void hostspace_fence(const DefaultHostExecutionSpace& exec, const char* msg);
```

Comment:
Are these changes required for this PR?

Reply:
This change is somewhat related to the philosophy of this PR (enhance when fencing is done, and how it is done, i.e. messages).

Reply:
"Somewhat related" is an issue here.  I am fine with the rest of the PR but not sure yet I agree with this change.

Reply:
I reverted the changes. But I think it would be good to create an issue somewhere to track what I wanted to improve (that basically calling `Kokkos::Impl::hostspace_fence` will just give to the profiling tools a message "Host fence", which is not very helpful).

Reply:
@dalg24 I reverted the changes, but then I also need to modify
```c++
//! Get fence event message for @c ZeroMemset.
template <typename execution_space>
std::string fence_event_message_for_zeromemset(const execution_space& space) {
  return std::string(space.name()).append(": ZeroMemset for");
}
```
back to
```c++
//! Get fence event message for @c ZeroMemset.
template <typename execution_space>
std::string fence_event_message_for_zeromemset(const execution_space& space) {
  return std::string("HostSpace fence");
}
```